src/main/java/core/basesyntax/Square.java

```diff
@@ -0,0 +1,27 @@
+package core.basesyntax;
+
+public class Square extends Figure {
+    private double side;
+
+    public Square(String color, double side) {
+        super(color);
+        this.side = side;
+    }
+
+    @Override
+    public String toString() {
+        return "Figure: square, area: " + calculateArea() + " sq. units, "
+                + "side: " + side + " units, "
+                + "color: " + getColor();
+    }
+
+    @Override
+    public double calculateArea() {
+        return side * side;
+    }
+
+    @Override
+    public void draw(Figure figure) {
+        System.out.println(figure);
+    }

```

Comment:
The 'draw' method should not be implemented in the 'Square' class since it's not conceptually related to the 'Square' class responsibilities. Consider moving it to a separate interface if multiple figures need a common drawing behavior.